### PR TITLE
C# xds interop tests: add support for xds splitting and routing

### PR DESCRIPTION
**Pull request by jtattermusch**
see b/161800492

Currently this PR is against the release branch (v1.31.x), but I can recreate against master if this is not necessary (I thought it was?).

### CODE REVIEW

**Comment by jtattermusch** [comment]
Adhoc test: sponge/daa76ab8-6fed-4768-b01d-46c65436af5c

File: tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh [hunk 1]
```diff
@@ -48,9 +48,14 @@ touch "$TOOLS_DIR"/src/proto/grpc/testing/__init__.py
...
+#
+# TODO(jtattermusch): remove "path_matching" and "header_matching" from
+# --test_case after they are added into "all".
 GRPC_VERBOSITY=debug GRPC_TRACE=xds_client,xds_resolver,xds_routing_lb,cds_lb,eds_lb,priority_lb,weighted_target_lb,lrs_lb "$PYTHON" \
   tools/run_tests/run_xds_tests.py \
-    --test_case=all \
+    --test_case="all,path_matching,header_matching" \
     --project_id=grpc-testing \
     --source_image=projects/grpc-testing/global/images/xds-test-server \
     --path_to_server_binary=/java_server/grpc-java/interop-testing/build/install/grpc-interop-testing/bin/xds-test-server \
```
Comment (stanley-cheung) [inline tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh:61]: Do we need to fix the `--client_cmd` (3 lines below this), to include these new parameters `{fail_on_failed_rpc} {rpcs_to_send} {metadata_to_send}`? https://github.com/grpc/grpc/blob/master/tools/internal_ci/linux/grpc_xds_bazel_test_in_docker.sh#L64-L66

Comment (ericgribkoff) [reply tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh:61]: `{rpcs_to_send}` and `{metadata_to_send}` are necessary here

Comment (menghanl) [reply tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh:61]: Yes, the parameters are necessary. And also need to change `--source_image` to `projects/grpc-testing/global/images/xds-test-server-2` (Append `-2`)

Comment (menghanl) [reply tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh:61]: @jtattermusch the PR is only missing the fixes in this file. Others look good.

**Comment by jtattermusch** [comment]
New adhoc run after applying the fixes: sponge/fa2369e7-915c-4638-ad4f-def83dcbddc9

File: tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh (hunk 1 above)
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_xds_csharp_test_in_docker.sh:61]: I fixed the cmdline args. Good catch, thanks!

**Comment by jtattermusch** [comment]
Added one more fix and a workaround for `--metadata=` argument (which doesn't get interpreted correctly by C# command line parser).
Another adhoc run:
https://sponge2.corp.google.com/f27e85b0-6d1d-489c-9475-fca5fd8340d9

**Comment by jtattermusch** [comment]
Adhoc test run is green: https://g3c.corp.google.com/results/invocations/f27e85b0-6d1d-489c-9475-fca5fd8340d9/targets

**Pull request merged**